operator-common/src/main/java/io/strimzi/operator/common/Util.java

```diff
@@ -858,4 +862,32 @@ public static boolean isMaintenanceTimeWindowsSatisfied(Reconciliation reconcili
     public static String encodeToBase64(String encode)  {
         return Base64.getEncoder().encodeToString(encode.getBytes(StandardCharsets.US_ASCII));
     }
+
+    /**
+     * Validate container's common compute resources.
+     * 
+     * @param resources Compute resources configuration.
+     * @param type Compute resource type (cpu or memory).
+     * @param owner Compute resource owner.
+     * 
+     * @return Error message or empty (no error found).
+     */
+    public static Optional<String> validateComputeResources(ResourceRequirements resources, String type, String owner) {
+        if (resources != null && resources.getRequests() != null && resources.getLimits() != null 
```

Comment:
I don't really get the `Arrays.asList("cpu", "memory").contains(type)` ... I would expect this to apply to all resource types as a general rule. If you want to use this only for `cpu` and `memory`, I'm fine. But it should reflect in the method name and you should maybe consider:
* Throwing some exception if wrong type is passed?
* Hide the type behind two public methods - one for CPU and on for Memory - calling thi private method with a given type

But I would prefer to just keep it generic and let it handle any type.